### PR TITLE
Improve tsc compatibility

### PR DESCRIPTION
**Pull request by abrkn**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
         "experimentalDecorators": true,
         // "strict": true,
         "strictNullChecks": true,
-        "strictFunctionTypes": false
+        "strictFunctionTypes": false,
+        "allowJs": true
```
Comment (johngrantuk) [inline tsconfig.json:16]: Thanks for your PR!
Could you explain why this change is useful/neccesary?

Comment (abrkn) [reply tsconfig.json:16]: As I write in the commit message of 167f151800dbb8c5429fc63dd7342a9a1e91ec55:

> Because `hardhat.config.js` is listed in `tsconfig.json`'s `files`,
`allowJs` must be enabled

Comment (johngrantuk) [reply tsconfig.json:16]: Have updated hardhat config so this should no longer be necessary.

Comment (abrkn) [reply tsconfig.json]: I've removed the commit

**Review by johngrantuk — APPROVED** [review]

**Comment by abrkn** [comment]
Done
